lib/routes/auth.js

```diff
@@ -0,0 +1,46 @@
+const router = require('express').Router();
+const User = require('../model/User');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const { registerValidation, loginValidation } = require('../routes/validation');
+
+router.post('/register', async (request, response) => {
+  const { error } = registerValidation(request.body);
+  if (error) return response.status(400).send(error.details[0].message);
+
+  const emailExists = await User.findOne({ email: request.body.email });
```

Comment:
You access `request.body.email` and `request.body.username` multiple times.
This is a good case for destructuring:
```js
const { email, username, password } = request.body;
```